[BD-13][BB-6209] refactor: deprecate node_path attribute of ModuleSystem

tecoholic opened the pull request:
## Description

The `node_path` attribute & constructor argument of the `ModuleSystem` is deprecated without any replacement service or fallback as there doesn't seem to be any core blocks using it.

~The CAPA modules and the blocks also reference to the `node_path` attribute, but are linked to the `LoncapaSystem`, which are not deprecated in this PR.~

**Update:** The PR removes all references to `node_path` as there doesn't seem to be any system using the value.


## Supporting information

- [Related Opencraft ticket](https://tasks.opencraft.com/browse/BB-6209)

## Testing instructions

- The XModule unit tests were confirmed as passing
- No core modules or XBlocks were found to be using the attribute. So no further testing was done.

## Deadline

"None"

tecoholic committed:
refactor: deprecate node_path attribute of ModuleSystem

The node_path attribute & constructor argument of the ModuleSyste is
deprecated without any replacement service or fallback as there doesn't
seem to be any core blocks using it.

The CAPA modules and the blocks also reference to the node_path
attribute, but are linked to the LoncapaSystem, which are not deprecated
in this commit.

openedx-webhooks commented:
Thanks for the pull request, @tecoholic! I've created [BLENDED-1234](https://openedx.atlassian.net/browse/BLENDED-1234) to keep track of it in Jira.   More details are on the [BD-13](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3062333478/BD-13+xModule+--+xBlock+Conversion+Phase+2) project page.


When this pull request is ready, tag your edX technical lead.


<!-- comment:welcome-blended -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

Agrendalath reviewed (changes requested):
@tecoholic,
> The CAPA modules and the blocks also reference to the node_path attribute, but are linked to the LoncapaSystem, which are not deprecated in this PR.

This is not correct. We can see [here](https://github.com/openedx/edx-platform/blob/0ef57eb136914a4dd0de1396095c80d8eddf73ac/common/lib/xmodule/xmodule/capa_module.py#L832) that the CAPA module retrieves the `node_path` from runtime (`self.runtime.node_path`) while creating the `LoncapaSystem`. If we remove the deprecation from `x_module.py`, then we will end up with the `AttributeError` exception while constructing a CAPA XBlock.

I'm curious if we need the `NODE_PATH` at all - I've just tried removing all references (including [these settings](https://github.com/openedx/edx-platform/blob/289e682b8f5fd4770aff5fe9147c365a5074a43e/lms/envs/common.py#L1090-L1098) and [this shim](https://github.com/openedx/edx-platform/blob/16fa97dde4560edb347a3743a9e7e345a6d74430/openedx/core/djangoapps/xblock/runtime/shims.py#L148-L156)), and everything was working correctly for me. Let's check [why it was added](https://github.com/openedx/edx-platform/commit/b9f2d4bc135b0b7e98354f1529b5907e2b913e17) (this is the commit from `git blame` of these vars - I only left it for reference, as I didn't dig deeper)  because it might be safe to simply delete all these references.

tecoholic commented:
@Agrendalath That's a good catch and I agree we can probably remove all the references as I don't see any use of the `node_path` in the platform base, only assignments to the variable. I wanted to remove the `LoncapaSystem` references as well, but let if untouched since I thought it was independent and not in the scope of the ticket, completely missed the `self.runtime.node_path` reference.

tecoholic committed:
refactor: remove all references to node_path

Removes the references to node_path from the LMS setting, the
LoncapaModuleSystem and the runtime shim.

tecoholic commented:
@Agrendalath Removed the references to `node_path` from all other places.

Agrendalath committed:
test: remove remaining node_path references

Agrendalath reviewed (approved):
:+1:

* [x] I tested this: checked that all references within `edx-platform` and its dependencies were not used; verified that the CAPA block works correctly after this change
* [x] I read through the code
* [x] I checked for accessibility issues: n/a
* [x] Includes documentation: n/a
* [x] I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository: n/a

Agrendalath commented:
@ormsbee, would you like to take a look at this before we merge it?

ormsbee commented on common/lib/xmodule/xmodule/x_module.py:
```diff
@@ -1609,6 +1609,18 @@ def filestore(self):
+
+        Deprecated.
+        """
+        warnings.warn(
+            'node_path is deprecated. Please use other methods of finding the node_modules location.',
+            DeprecationWarning, stacklevel=3
```
@jristau1984: After this merges and deploys, please try searching for this log message in Splunk. The only use I recall that this functionality ever had was for some early Berkeley courses that had to be explicitly put on an allow list to directly call node in some of their problem code. My recollection is that this functionality is long gone and those courses are not even accessible any longer (as they were Old Mongo courses). But if there are other instances in the code that try to use the node_path in this way, it should show up in the logs.

openedx-webhooks commented:
@tecoholic 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-platform&entry.867055334=https://github.com/openedx/edx-platform/pull/30447&entry.1484655318=https://github.com/tecoholic&entry.752974735=2022-05-20+05:50&entry.1917517419=2022-06-08+16:06&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
